language/move-lang/src/typing/translate.rs

```diff
@@ -119,7 +124,7 @@ fn check_primitive_script_arg(context: &mut Context, mloc: Loc, idx: usize, ty:
         result.is_ok()
     };
     if is_signer_ref {
-        if idx == 0 {
+        if !*seen_non_signer {
```

Comment:
test for this?

Reply:
The previous tests are testing for this (unless I'm reading them wrong). Happy to add a couple more.

Reply:
Didn't see the move tests being updated (maybe the push for that happened after I commented? unsure)
Regardless, the tests look fine

Reply:
Looks like we need to update the error message